src/helpers/balances.js

```diff
@@ -116,6 +124,14 @@ exports.getBalancesFromPayments = (payment, tppList) => {
   return bill;
 };
 
+const getRemainingPaymentType = (clients, paymentType) => !clients.some((cl) => {
```

Comment:
pas fan du nom 'PaymentType', vous auriez mieux ?

Reply:
Ici pareil comme c'est un boolean, le get n'est pas forcement parlant.
que penses-tu de `isDocRemaining(clients, doc)` ?

Reply:
je me dis que pour trouver un nom c'est cool de revenir au pourquoi de cette méthode : 
- dans la méthode `getBalances`, on recupère les factures, avoirs, paiements pour calculer les balance clients
- dans un monde idéal, tout part de la facture - c'est ce qu'on retrouve dans cette partie
```
for (const bill of bills) {
    clients.push({ ...bill._id });
    balances.push(exports.getBalance(bill, customerCNAggregation, tppCNAggregation, payments, tppList));
  }
```
on boucle sur les couple client/beneficiaire des factures, et pour chacun, on calcule leur balance.
le monde idéal termine ici.

CE-PEN-DANT ...

- on peut se retrouver dans des cas ou, pour un couple client/beneficiaire, on n'a pas de facture mais on a un avoir et un paiment. c'est l'objectif de ce bout de code et de la boucle for qui suit : Recuperer tous les avoirs qui n'ont pas ete traité par les factures et calculer la balances pour ces clients la
```
const remainingCreditNotes = [...customerCNAggregation, ...tppCNAggregation].filter(cn => getRemainingPaymentType(clients, cn));
```
c'est pour ca qu'on remplit cette variable `clients` pour garder en memoire les couples qu'on a deja géré `clients.push({ ...cn._id });`

une fois qu'on a geré les avoirs on regarde ls cas ou on n'a ni facture ni avoir mais qu'on a un paiement

Reply:
du coup pour revenir a la fonction ci dessus,
- je me dis que c'est cool si elle renvoit plutot `clients.some` et non `!clients.some`
- que pensez vous du nom `isAlreadyUsed` ?

Reply:
Plutot que used que penses-tu de isAlreadyHandled(ou processed ou treated) ?

Reply:
j'ai bien le processed :)

Reply:
done !